vowpalwabbit/epsilon_reduction_features.h

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+
+#pragma once
+
+namespace VW
+{
+namespace cb_explore_adf
+{
+namespace greedy
+{
+struct reduction_features
+{
+  float epsilon;
+
+  reduction_features() { epsilon = -1.f; }
```

Comment:
Wondering if -1.f should be a constant (INVALID_EPSILON ?) and then reference that? I realize this is not standard practice in VW :).

Reply:
Another idea might be to add a is_valid_epslion() method to reduction features.  This way no one needs to know about it.

Reply:
Yea good point, I added in is_valid_epsilon()